proof/access-control/AARCH64/ArchAccess_AC.thy

```diff
@@ -0,0 +1,261 @@
+
+named_theorems Access_AC_assms
+
+lemma acap_class_reply[Access_AC_assms]:
+  "acap_class acap \<noteq> ReplyClass t"
+  by (cases acap; simp)
+
+lemma arch_troa_tro_alt[Access_AC_assms, elim!]:
+  "arch_integrity_obj_atomic aag subjects l ko ko'
+   \<Longrightarrow> arch_integrity_obj_alt aag subjects l ko ko'"
+  by (fastforce elim: arch_integrity_obj_atomic.cases intro: arch_integrity_obj_alt.intros)
+
+lemma clear_asidpool_trans[elim]:
+  "\<lbrakk> asid_pool_integrity subjects aag pool pool';
+     asid_pool_integrity subjects aag pool' pool'' \<rbrakk>
+     \<Longrightarrow> asid_pool_integrity subjects aag pool pool''"
+  unfolding asid_pool_integrity_def by metis
+
+lemma cap_asid'_member[simp]:
+  "asid \<in> cap_asid' cap = (\<exists>acap. cap = ArchObjectCap acap \<and> asid \<in> acap_asid' acap)"
+  by (cases cap; clarsimp)
+
+lemma clas_caps_of_state[Access_AC_assms]:
+  "\<lbrakk> caps_of_state s slot = Some cap; pas_refined aag s \<rbrakk>
+     \<Longrightarrow> cap_links_asid_slot aag (pasObjectAbs aag (fst slot)) cap"
+  apply (clarsimp simp: cap_links_asid_slot_def label_owns_asid_slot_def pas_refined_def)
+  apply (drule state_asids_to_policy_aux.intros)
+   apply assumption
+  apply (blast dest: state_asids_to_policy_aux.intros)
+  done
+
+lemma arch_tro_alt_trans_spec[Access_AC_assms]:
+  "\<lbrakk> arch_integrity_obj_alt aag subjects l ko ko';
+     arch_integrity_obj_alt aag subjects l ko' ko'' \<rbrakk>
+     \<Longrightarrow> arch_integrity_obj_alt aag subjects l ko ko''"
+  by (fastforce simp: arch_integrity_obj_alt.simps)
+
+end
+
+
+global_interpretation Access_AC_1?: Access_AC_1
+proof goal_cases
+  interpret Arch .
+  case 1 show ?case
+    by (unfold_locales; fact Access_AC_assms)
+qed
+
+
+context Arch begin global_naming AARCH64
+
+lemma auth_ipc_buffers_tro[Access_AC_assms]:
+  "\<lbrakk> integrity_obj_state aag activate subjects s s';
+     x \<in> auth_ipc_buffers s' p; pasObjectAbs aag p \<notin> subjects \<rbrakk>
+     \<Longrightarrow> x \<in> auth_ipc_buffers s p "
+  by (drule_tac x = p in spec)
+     (erule integrity_objE;
+      fastforce simp: tcb_states_of_state_def get_tcb_def auth_ipc_buffers_def
+               split: cap.split_asm arch_cap.split_asm if_split_asm bool.splits)
+
+lemma trasids_trans[Access_AC_assms]:
+  "\<lbrakk> (\<forall>x a. integrity_asids aag subjects x a s s');
+     (\<forall>x a. integrity_asids aag subjects x a  s' s'') \<rbrakk>
+     \<Longrightarrow> (\<forall>x a. integrity_asids aag subjects x a s s'')"
+  by clarsimp metis
+
+lemma integrity_asids_refl[Access_AC_assms, simp]:
+  "integrity_asids aag subjects x a s s"
+  by simp
+
+lemma integrity_asids_update_autarch[Access_AC_assms]:
+  "\<lbrakk> \<forall>x a. integrity_asids aag {pasSubject aag} x a st s; is_subject aag ptr \<rbrakk>
+     \<Longrightarrow> \<forall>x a. integrity_asids aag {pasSubject aag} x a st (s\<lparr>kheap := (kheap s)(ptr \<mapsto> obj)\<rparr>)"
```

Comment:
I'm guessing all of these lemmas with old-style `==>` indents are only moved from elsewhere?

Reply:
They are Tom-style indents, not old style indents. I don't think anybody else has used them :-)

Reply:
Right, those are not new then, fine to leave as is